resource/ui/ts/models/log.ts

```diff
@@ -0,0 +1,149 @@
+        var tdStyleEvenFirst = "font-family:Arial, sans-serif;font-size:14px;padding:10px 5px;border-style:solid;border-width:1px;overflow:hidden;word-break:normal;border-color:#ccc;color:#333;background-color:#efefef;text-align:center";
+        var tdStyleEven = "font-family:Arial, sans-serif;font-size:14px;padding:10px 5px;border-style:solid;border-width:1px;overflow:hidden;word-break:normal;border-color:#ccc;color:#333;background-color:#fff;text-align:center";
+
+        export class Entries {
+
+            startTime = Utils.chainProp(this, <number>null);
+            endTime = Utils.chainProp(this, <number>null);
+            max = Utils.chainProp(this, <number>null);
+            level = Utils.chainProp(this, <string>null);
+
+            private _url():string {
+                var url = "/_status/local/log";
+                if (this.level() != null) {
+                    url += "/" + this.level();
+                }
+                url += "?";
+                if (this.startTime() != null) {
+                    url += "startTime=" + this.startTime().toString() + "&";
+                }
+                if (this.endTime() != null) {
+                    url += "entTime=" + this.endTime().toString() + "&";
+                }
+                if (this.max() != null) {
+                    url += "max=" + this.max().toString() + "&";
+                }
+                return url;
+            }
+
+            private _data = new Utils.QueryCache((): promise<Proto.LogEntry[]> => {
+                return m.request({ url: this._url(), method: "GET", extract: nonJsonErrors })
+                    .then((results: LogResponseSet) => {
+                        return results.d;
+                    });
+            })
+
+            public Refresh() {
+                this._data.refresh();
+            }
+
+            public Entries(): Proto.LogEntry[] {
+                return this._data.result();
+            }
+
+            private static _formatTags = new RegExp("%s|%d|%v|%+v", "gi")
+            private static _FormatMessage(entry: Proto.LogEntry): string {
+                var i = -1;
+                return entry.format.replace(Entries._formatTags, function(tag) {
+                    i++;
+                    if (entry.args.length > i) {
+                        return entry.args[i].str;
+                    } else {
+                        return "";
+                    }
+                });
+            }
+
+            private static _EntryRow(entry: Proto.LogEntry, count: number): _mithril.MithrilVirtualElement {
+                var dstyle, countStyle: string;
+                if (count % 2 == 0) {
+                    countStyle = tdStyleEvenFirst;
+                    dstyle = tdStyleEven;
+                } else {
+                    countStyle = tdStyleOddFirst;
+                    dstyle = tdStyleOdd;
+                }
+                return m("tr", [
```

Comment:
I'll move this code into the view.  And yay tables!
I'm fine turning these into some nice css based tables instead, I'm just not going to spend the time working on that yet.